### DISCUSSION TITLE
How to get typed getStaticProps with locale value ?

### DISCUSSION THREAD

**Discussion by MoOx** [amannn/next-intl#165] [Q&A] [answered]
I am new to TypeScript and would like to know how I can get this validated by TS.

```ts
export const getStaticProps: GetStaticPaths = async ({ locale }) =>{
  return {
    props: {
      messages: {
        ...require(`locales/${locale}.json`),
      },
    },
  };
}
```

`locale` property throws an error `Property 'locale' does not exist on type 'GetStaticPathsContext'`.

Is there a trick to let know `GetStaticPaths` that there is additional values incoming ?

<img width="1014" alt="image" src="https://user-images.githubusercontent.com/157534/211821671-922a8ff1-fb6b-46bc-bc7f-b19faf54883c.png">

**Comment by amannn** [comment]
Have a look at the example that is linked to in the docs, this should help:

https://github.com/amannn/next-intl/blob/d5d894fc0fd1c4ec71ce470fc089cb9af8a7e224/packages/example/src/pages/index.tsx#L17-L23

**Reply by MoOx** [reply]
Some types are missing here.

**Reply by MoOx** [reply]
But a piece is good :)

**Comment by MoOx** [comment, marked as answer]
One mistake: `getStaticProps: GetStaticPaths` when it should be `getStaticProps: GetStaticProps`.
Then like @amannn mentioned `({locale}: GetStaticPropsContext)` to define arguments type.
